vision/detect_crop_hints.rb

```diff
@@ -12,22 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def detect_crop_hints project_id:, image_path:
+require "uri"
+
+def detect_crop_hints image_path:
   # [START vision_crop_hint_detection]
-  # project_id = "Your Google Cloud project ID"
   # image_path = "Path to local image file, eg. './image.png'"
-  
+
   require "google/cloud/vision"
 
-  vision = Google::Cloud::Vision.new project: project_id
-  image  = vision.image image_path
+  image_annotator = Google::Cloud::Vision::ImageAnnotator.new
 
-  crop_hints = image.crop_hints
+  content = File.binread image_path
+  image = { content: content }
+  feature = { type: :CROP_HINTS }
```

Comment:
You shouldn't have to manually set the [`Feature.Type`](https://cloud.google.com/vision/docs/reference/rpc/google.cloud.vision.v1#google.cloud.vision.v1.Feature.Type) by hand, there should be a `#crop_hints_detection` or similar `#[FEATURE_NAME]_detect` helper method